File: Part 2 - Regression/Section 4 - Simple Linear Regression/simple_LR.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Tue Mar  3 11:47:48 2020

@author: macbook
"""
#Importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

#Importing dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:,1].values

#Splitting the dataset into training set and testing set
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=1/3,random_state=0)

#Fitting SLR to Training Set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,y_train)

#Predicting the Test set results
y_pred = regressor.predict(X_test)

#Visualising the Training Set results 
plt.scatter(X_train,y_train,color='red')
plt.plot(X_train,regressor.predict(X_train),color='blue')
plt.title('Salary vs Experience (Training Set)')
plt.xlabel('Years of experience')
plt.ylabel('Salary')
plt.show()

#Visualising the Testing Set results 
plt.scatter(X_test,y_test,color='red')
plt.plot(X_train,regressor.predict(X_train),color='blue')
plt.title('Salary vs Experience (Testing Set)')
plt.xlabel('Years of experience')
plt.ylabel('Salary')
plt.show()

File: Part 2 - Regression/Section 4 - Simple Linear Regression/simple_LR.R
#Simple Linear Regression

#Importing dataset
dataset = read.csv('Salary_Data.csv')

#Spliting dataset to training sets and testing sets
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary,SplitRatio=2/3)
training_set = subset(dataset,split == TRUE)
test_set = subset(dataset,split == FALSE)

#Fitting SLR to Training Set
regressor = lm(formula = Salary ~ YearsExperience,data = training_set)

#Predicting the Test set results
y_pred = predict(regressor, newdata = test_set)

#Visulaizing the training set results
#install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience , y = training_set$Salary),
             color = 'red') +
  geom_line(aes(x = training_set$YearsExperience , y = predict(regressor, newdata = training_set)),
            color = 'blue') +
  ggtitle('Salary vs Experience (Training Set)') +
  xlab('Years of experience') +
  ylab('Salary')


#Visualising the Testing Set results 
library(ggplot2)
ggplot() +
  geom_point(aes(x = test_set$YearsExperience , y = test_set$Salary),
             color = 'red') +
  geom_line(aes(x = training_set$YearsExperience , y = predict(regressor, newdata = training_set)),
            color = 'blue') +
  ggtitle('Salary vs Experience (Test Set)') +
  xlab('Years of experience') +
  ylab('Salary')
